src/ln/channelmanager.rs

```diff
@@ -466,6 +476,30 @@ macro_rules! try_chan_entry {
 				}
 				return Err(MsgHandleErrInternal::from_finish_shutdown(msg, channel_id, chan.force_shutdown(), $self.get_channel_update(&chan).ok()))
 			},
+			Err(ChannelError::Wait { msg, update }) => {
+				log_error!($self, "Channel {} need to be shutdown but closing transactions not broadcast due to {}", log_bytes!($entry.key()[..]), msg);
+				let (channel_id, mut chan) = $entry.remove_entry();
+				if let Some(short_id) = chan.get_short_channel_id() {
+					$channel_state.short_to_id.remove(&short_id);
+				}
+				if let Some(update) = update {
+					if let Err(e) = $self.monitor.add_update_monitor(update.get_funding_txo().unwrap(), update) {
+						match e {
+							// Our local state is rotten as we're fallen-behind and we can't send our updates
+							// to watchtowers to retrieve our funds from remote commitment transaction.
+							// We close channnel as there is no hope to get back something..
+							ChannelMonitorUpdateErr::PermanentFailure => {
+								return Err(MsgHandleErrInternal::from_finish_shutdown(msg, channel_id, chan.force_shutdown(), $self.get_channel_update(&chan).ok()))
```

Comment:
I dont think you should handle the force_shutdown result differently here?

Reply:
Okay because we may still use our fresh in-memory ChannelMonitor with remote per_commitment_point to get back the to_remote output ?